lib/galaxy/webapps/tool_shed/model/__init__.py

```diff
@@ -246,10 +246,12 @@ def get_type_class( self, app ):
         return app.repository_types_registry.get_class_by_label( self.type )
 
     def get_tool_dependencies( self, changeset_revision ):
+        repo = hg.repository( ui.ui(), self.repo_path( app ) )
```

Comment:
Also, `repo = hg_util.get_repo_for_repository( app, repository=self )` is probably cleaner, but would require to add `hg_util` to `from tool_shed.util import metadata_util`